mamutils/sendreceivemam.go

```diff
@@ -0,0 +1,87 @@
+/*
+MIT License
+Copyright (c) 2017 Harry Boer
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+//Package mamutils is a utility to create Masked Authenticated Messages (MAM's)
+package mamutils
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/iotaledger/giota"
+)
+
+//ToMAMTrytes checks its validity and casts to giota.Trytes.
+func ToMAMTrytes(t string) (tr giota.Trytes) {
+
+	trytes := ""
+	TryteValues := giota.TryteAlphabet
+
+	for _, number := range t {
+		if number > 255 {
+			//make it a space
+			number = 32
+		}
+		firstValue := number % 27
+		secondValue := (number - firstValue) / 27
+		trytesValue := string(TryteValues[firstValue]) + string(TryteValues[secondValue])
+		trytes = trytes + trytesValue
+	}
+
+	newTrytes := giota.Trytes(trytes)
+
+	return newTrytes
+}
+
+//FromMAMTrytes converts the MAM from giota.Trytes to a readable string
+func FromMAMTrytes(inputTrytes giota.Trytes) string {
+
+	outputString := ""
+	TryteValues := giota.TryteAlphabet
+
+	// Check if input is an even number of giota.Trytes
+	err := IsValidTrytes(inputTrytes)
+	if err != nil {
+		fmt.Println("Error: ", err)
```

Comment:
I will do the update later. Unfortunately they took the message board off-line, so I can only check it via the CLI, but that isn't good enough. Next week there will be another IOTA meet-up in Amsterdam, where I will ask to let the message board up for development.
Why aren't you coming to the meet-up? Last time they mentioned that it's the largest meetup in Europe.